plugins/modules/aci_l4l7_policy_based_redirect_destination.py

```diff
@@ -0,0 +1,404 @@
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="vnsSvcRedirectPol",
+            aci_rn="svcCont/svcRedirectPol-{0}".format(policy),
+            module_object=policy,
+            target_filter={"name": policy},
+        ),
+        subclass_2=dict(
+            aci_class=aci_class,
+            aci_rn=aci_rn,
+            module_object=module_object,
+            target_filter=target_filter,
+        ),
+        child_classes=child_classes,
+    )
+    aci.get_existing()
+
+    if state == "present":
+        if destination_type == "l1/l2" and additional_ip is not None:
+            aci.fail_json(msg="You cannot provide an additional_ip when configuring an l1/l2 destination")
+        elif destination_type == "l3" and (logical_device, concrete_device, concrete_interface) != (None, None, None):
+            aci.fail_json(msg="You cannot provide a logical_device, concrete_device or concrete_interface when configuring an l3 destination")
+        elif destination_type == "l1/l2" and (logical_device, concrete_device, concrete_interface) == (None, None, None):
+            aci.fail_json(msg="You must provide a logical_device, concrete_device and concrete_interface when configuring an l1/l2 destination")
+        elif destination_type == "l1/l2" and ip is not None:
+            aci.fail_json(msg="You cannot provide an ip when configuring an l1/l2 destination")
+        child_configs = []
+        if destination_type == "l1/l2":
+            child_configs = [
+                {
+                    "vnsRsToCIf": {
+                        "attributes": {"tDn": "uni/tn-{0}/lDevVip-{1}/cDev-{2}/cIf-[{3}]".format(tenant, logical_device, concrete_device, concrete_interface)}
+                    }
+                }
+            ]
+        if health_group is not None:
+            health_group_tdn = "uni/tn-{0}/svcCont/redirectHealthGroup-{1}".format(tenant, health_group)
+            child_configs.append({redirect_hg_class: {"attributes": {"tDn": health_group_tdn}}})
+        else:
+            health_group_tdn = None
+        if isinstance(aci.existing, list) and len(aci.existing) > 0:
+            for child in aci.existing[0].get(aci_class, {}).get("children", {}):
+                if child.get(redirect_hg_class) and child.get(redirect_hg_class).get("attributes").get("tDn") != health_group_tdn:
+                    child_configs.append(
+                        {
+                            redirect_hg_class: {
+                                "attributes": {
+                                    "dn": child.get(redirect_hg_class).get("attributes").get("dn"),
+                                    "status": "deleted",
+                                }
+                            }
+                        }
+                    )
+        aci.payload(
+            aci_class=aci_class,
+            class_config=dict(ip=ip, mac=mac, destName=destination_name, podId=pod_id, ip2=additional_ip),
```

Comment:
format on different lines? run black after
```suggestion
            class_config=dict(ip=ip, mac=mac, destName=destination_name, podId=pod_id, ip2=additional_ip,),
```